### PR TITLE
Add BookingSG ErrorDisplay assets

### PR DESCRIPTION
**Pull request by nobeltws**
**Changes**
- Added error display data for BSG
- Added function to get error display based on theme (e.g. base/bookingsg)
- Updated mock for error display tests

- [delete] branch

**Changelog entry**
- Added error display data for BSG

**Additional information**

- You may refer to this [BOOKINGSG-3895](https://jira.ship.gov.sg/browse/BOOKINGSG-3895)
- This was a requirement from Michelle so that we have copy and illustrations for BSG directly from DS (Illustrations are not done yet, will raise another PR to update illustrations once its done)

### CODE REVIEW

File: src/error-display/error-display-data.tsx [hunk 1]
```diff
@@ -235,3 +236,133 @@ export const ERROR_DISPLAY_DATA = new Map<
...
+            title: "Browser not supported",
+            description:
+                "BookingSG works best with the latest version of Chrome, Edge, Firefox or Safari. If you continue with your browser, you may run into problems.",
+        },
+    ],
+]);
+
+export const errorDisplayData = (
+    type: ErrorDisplayType,
+    colorScheme: ColorScheme = "base"
```
Comment (keithtxw) [inline src/error-display/error-display-data.tsx:361]: `colorScheme` is not a correct key to use in this case. Pending discussion outcome

**Review by keithtxw — CHANGES_REQUESTED** [review]
Also missing stories to document the BSG assets.

**Comment by keithtxw** [comment]
Hold for now. Pending change in theme to accommodate to this use case

File: src/error-display/error-display.tsx [hunk 2]
```diff
@@ -26,7 +28,11 @@ export const ErrorDisplay = ({
     // =============================================================================
     // CONST, STATE, REF
     // =============================================================================
-    const defaultAssets = ERROR_DISPLAY_DATA.get(type);
+    const theme = useTheme();
+    const defaultAssets = errorDisplayData(
+        type,
+        (theme as unknown as ThemeSpec).resourceScheme
```
Comment (keithtxw) [inline src/error-display/error-display.tsx:34]: i think i've already typed  styled-components `DefaultTheme`. You should be able to access the `resourceScheme` without typecasting `theme`

**Review by keithtxw — CHANGES_REQUESTED** [review]
Will look to refactor the stories

File: src/error-display/error-display.tsx (hunk 2 above)
Comment (nobeltws) [reply src/error-display/error-display.tsx:34]: got it! will remove the type casting here. thanks!

**Comment by nobeltws** [comment]
Closing the PR since error page does not come with default copy anymore

**Pull request closed without merging**